fathom/src/core/semantics.rs

```diff
@@ -8,11 +8,35 @@ use std::sync::Arc;
 use crate::alloc::SliceVec;
 use crate::core::{Const, EntryInfo, Prim, Term, UIntStyle};
 use crate::env::{EnvLen, GlobalVar, SharedEnv, SliceEnv};
+use crate::source::Span;
 use crate::StringId;
 
 /// Atomically reference counted values. We use reference counting to increase
 /// the amount of sharing we can achieve during evaluation.
-pub type ArcValue<'arena> = Arc<Value<'arena>>;
+pub type ArcValue<'arena> = SpanValue<'arena>;
```

Comment:
> Is there any reason where you chose This, as opposed to `Arc<SpanValue<'arena>>`?

Yes, my thinking was that this allows `Values` to be reused/shared but if we put the span inside the Arc then they can only shared if they are from the same source location. This came up when adding a span for `fail`. Since `fail` refers to the same value in the environment I needed a way to attach source location to it when it was encountered. Does that make sense? It's also possible that I've missed something.

Reply:
Maybe `fail` could take a source location as part of the primitive? 🤔 eg.

```
SourceSpan : Type
fail : SourceSpan -> Format
```

Oh wait, this would only work for `reported_error`s as those are only even created by the elaborator. You need to make fail formats as a programmer… if we had glued eval we might be able to take it from glued thing. Hmm, need to think. Anyway, it's ok to keep pushing past this for now, but I think ultimately I _might_ prefer the `Arc<SpanValue<'arena>>` order.